src/utils/session-core.js

```diff
@@ -699,6 +703,67 @@ export const updateUserActivity = async (session) => {
   Functions here deal primarily with user inbox on PASS
 */
 
+/**
+ * Function that gets a list of users from their Solid Pod stored inside the
+ * Solid container named User
+ *
+ * @memberof utils
+ * @function getInboxMessageTTL
+ * @param {Session} session - Solid's Session Object {@link Session}
+ * @param {inboxListObject[]} inboxList - List of inbox messages
+ * @returns {Promise} Promise - An array of users from their Pod into PASS, if
+ * users list exist
+ */
+
+export const getInboxMessageTTL = async (session, inboxList) => {
+  const inboxContainerUrl = getContainerUrl(session, 'Inbox', 'self-fetch');
+  let messageList = [];
+  try {
+    const solidDataset = await getSolidDataset(inboxContainerUrl, {
+      fetch: session.fetch
+    });
+    const ttlFileThing = getThingAll(solidDataset);
+    const allMessageThing = ttlFileThing.filter((thing) => thing.url.slice(-3).includes('ttl'));
+
+    // Early return if length of inbox in both PASS and Solid is the same
+    if (allMessageThing.length === inboxList.length) {
+      return inboxList;
+    }
+
+    try {
+      const promises = allMessageThing.map(async (messageTTL) => {
+        const messageDataset = await getSolidDataset(messageTTL.url, { fetch: session.fetch });
+
+        const messageTTLThing = getThingAll(messageDataset);
+
+        // Get data related to #message
+        const messageThing = messageTTLThing.find((thing) => thing.url.includes('#message'));
+        const message = getStringNoLocale(messageThing, RDF_PREDICATES.message);
+        const title = getStringNoLocale(messageThing, RDF_PREDICATES.title);
+        const uploadDate = getDatetime(messageThing, RDF_PREDICATES.uploadDate);
+
+        // Get data related to #sender
+        const senderThing = messageTTLThing.find((thing) => thing.url.includes('#sender'));
+        const sender = getStringNoLocale(senderThing, RDF_PREDICATES.sender);
+
+        // Get data related to #sender
```

Comment:
I think `#sender` should be `#recipient`

Reply:
I'll leave the functionality of this file to be reviewed by somebody more familiar with it (session-core.js) 

I'm getting a little lost in here 😅

I'm unsure about the `catch`es specifically.

Reply:
The first try/catch tries to see if they could find a user with the username being sent from PASS. If the user exists on the Solid Identity Provider (solidcommunity.net in this case), it'll set recipientName to the user's actual name, else it'll crash out with an error message that state the username does not exist.

Second try/catch tries to see if they could send a message to both the sender's and the recipient's inbox. If the inboxes exist, it'll create a TTL file in both inboxes. If either one of those inboxes does not exist, it'll crash out with an error message that states that at least one of the inboxes does not exist.

Reply:
@benzelcurry @leekahung thoughts on my first comment here? line 749...

Reply:
Oh! I see it now. Good catch! You're right,  it supposed to be recipient, not sender on Line 749

Reply:
Corrected this from `#sender` to `#recipient`